dfastmi/gui/dialog_model.py

```diff
@@ -0,0 +1,126 @@
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with this library; if not, see <http://www.gnu.org/licenses/>.
+
+contact: delft3d.support@deltares.nl
+Stichting Deltares
+P.O. Box 177
+2600 MH Delft, The Netherlands
+
+All indications and logos of, and references to, "Delft3D" and "Deltares"
+are registered trademarks of Stichting Deltares, and remain the property of
+Stichting Deltares. All rights reserved.
+
+INFORMATION
+This file is part of D-FAST Morphological Impact: https://github.com/Deltares/D-FAST_Morphological_Impact
+"""
+from configparser import ConfigParser, SectionProxy
+from typing import List, Optional
+import dfastmi
+from dfastmi.io.AReach import AReach
+from dfastmi.io.Branch import Branch
+from dfastmi.io.RiversObject import RiversObject
+from dfastmi.io.ConfigFileOperations import ConfigFileOperations, check_configuration
+
+
+class DialogModel:
+    config : ConfigParser = None
+    section : SectionProxy = None
+    def __init__(self, rivers_configuration: RiversObject, config_file: Optional[str] = None):
+        self.rivers = rivers_configuration
+        
+        if config_file:
+            self.load_configuration(config_file)
+
+
+    def load_configuration(self, filename: str) -> None:
+        try:
+            self.config = ConfigFileOperations.load_configuration_file(filename)
+        except:
+            if filename != "dfastmi.cfg":
+                return False                
+            return True
+    
+        self.section = self.config["General"]
+
+    def check_configuration(self, branch : Branch, reach: AReach, reference_files: List, measure_files : List) -> bool :
+        config = self.get_configuration(branch, reach, reference_files, measure_files)
+        return check_configuration(self.rivers, config)
+    
+
+    def run_analysis(self) -> bool:
+        # Logic to run analysis based on configuration
+        """
+        Run the D-FAST Morphological Impact analysis based on settings in the GUI.
+
+        Arguments
+        ---------
+        None
+        """
+        try:
+            success = dfastmi.batch.core.batch_mode_core(self.rivers, False, self.config)
+        except:
+            success = False
+        return success
+    
+    
+    def get_configuration(self, branch : Branch, reach: AReach, reference_files: List, measure_files : List) -> ConfigParser:
+        """
+        Extract a configuration from the GUI.
+
+        Arguments
+        ---------
+        None
+
+        Returns
+        -------
+        config : Optional[configparser.ConfigParser]
+            Configuration for the D-FAST Morphological Impact analysis.
+        """
+        config = ConfigParser()
+        config.optionxform = str
+        config.add_section("General")
+        config["General"]["Version"] = "2.0"
```

Comment:
should this be "2.0" or Version("2.0")?

Reply:
"2.0" In the config parser object you can only write strings.